portality/api/v1/bulk/articles.py

```diff
@@ -31,39 +31,43 @@ def create_swag(cls):
     def create(cls, articles, account):
         # we run through create twice, once as a dry-run and the second time
         # as the real deal
-        for a in articles:
-            ArticlesCrudApi.create(a, account, dry_run=True)
-        # no formatting issues, the JSON of each article complies with required format at this point
+        new_articles = []
 
         ids = []
         dois_seen = set()
         fulltext_urls_seen = set()
         for a in articles:
             duplicate = False  # skip articles if their fulltext URL or DOI is the same
 
-            for i in a['bibjson']['identifier']:
-                if i['type'] == 'doi':
-                    if i['id'] in dois_seen:
-                        duplicate = True
-                        break
-                    dois_seen.add(i['id'])
+            for i in a.get('bibjson', {}).get('identifier', []):
+                if i.get('type', '') == 'doi':
+                    if 'id' in i:
+                        if i['id'] in dois_seen:
+                            duplicate = True
+                            break
+                        dois_seen.add(i['id'])
 
             if duplicate:
                 continue
 
-            for l in a['bibjson']['link']:
-                if l['type'] == 'fulltext':
-                    if l['url'] in fulltext_urls_seen:
-                        duplicate = True
-                        break
-                    fulltext_urls_seen.add(l['url'])
+            for l in a.get('bibjson', {}).get('link', []):
+                if l.get('type', '') == 'fulltext':
+                    if 'url' in l:
+                        if l['url'] in fulltext_urls_seen:
+                            duplicate = True
+                            break
+                        fulltext_urls_seen.add(l['url'])
 
             if duplicate:
                 continue
 
-            n = ArticlesCrudApi.create(a, account)
+            n = ArticlesCrudApi.create(a, account, dry_run=True)
```

Comment:
I did locally, though I'm afraid I don't have the setup to easily re-try this at the moment. It was exactly 2x faster (relative to the same machine). That means, I believe, 24s instead of 48s on the machine I tried it on.